server/server.go

```diff
@@ -165,18 +167,28 @@ func (server *Server) firstTimeInit(configuration *config.Config) error {
 	return nil
 }
 
-func (server *Server) initHttp(configuration *config.Config) {
+func (server *Server) initHttp(configuration *config.Config) error {
 	server.useHttp = true
 	httpPort := configuration.HttpConfig.Port
+	if err := util.CheckPort(httpPort); err != nil {
+		return err
+	}
+
 	server.httpPort = DefaultHttpPort
 	if httpPort != 0 {
```

Comment:
so what you suggest is not using default port at all and using the value from the config file always?